resilience4j-bulkhead/src/main/java/io/github/resilience4j/bulkhead/internal/SemaphoreBulkhead.java

```diff
@@ -71,73 +73,88 @@ public SemaphoreBulkhead(String name) {
     /**
      * Create a bulkhead using a configuration supplier
      *
-     * @param name the name of this bulkhead
+     * @param name           the name of this bulkhead
      * @param configSupplier BulkheadConfig supplier
      */
     public SemaphoreBulkhead(String name, Supplier<BulkheadConfig> configSupplier) {
         this(name, configSupplier.get());
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void changeConfig(final BulkheadConfig newConfig) {
+        synchronized (configChangesLock) {
+            int delta =  newConfig.getMaxConcurrentCalls() - config.get().getMaxConcurrentCalls();
+            if (delta < 0) {
+                semaphore.acquireUninterruptibly(-delta);
+            } else if (delta > 0) {
+                semaphore.release(delta);
```

Comment:
Original permits in Semaphore constructor it is only the `initial` number of permits available.
There is actually no limit for them. Moreever it can even be negative number initially) In this case specified number of releases must occur before anyone can get permission.

Reply:
Yeah, I remember now that it was up to the client to make the cap actually hold & be consistent.